File: src/Backend/Services/Catalog/Data.SQL/Types/Topic/TopicTypeEntity.cs
// Copyright (c) 2023 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

namespace Crib2023.Backend.Services.Catalog.Data.SQL.Types.Topic;

/// <summary>
/// Сущность типа "Тема".
/// </summary>
public class TopicTypeEntity
{
    #region Properties

    /// <summary>
    /// Идентификатор.
    /// </summary>
    public long Id { get; set; }

    /// <summary>
    /// Название.
    /// </summary>
    public string Name { get; set; } = "";

    /// <summary>
    /// Идентификатор родителя.
    /// </summary>
    public long? ParentId { get; set; }

    /// <summary>
    /// Глобальный идентификатор строки.
    /// </summary>
    public Guid RowGuid { get; set; } = Guid.NewGuid();

    #endregion Properties
}

File: src/Frontend/React/TypeScript/App/src/data/Types/Topic/TopicTypeEntity.ts
export interface TopicTypeEntity {
  id: number;
  name: string;
  parentId: number;
  rowGuid?: string;
}

export function createTopicTypeEntity (options?: Partial<TopicTypeEntity>): TopicTypeEntity {
  return {
    id: options?.id ?? 0,
    name: options?.name ?? '',
    parentId: options?.parentId ?? 0,
    rowGuid: options?.rowGuid ?? ''
  };
}
